crates/oxc_linter/src/rules/jsdoc/empty_tags.rs

```diff
@@ -0,0 +1,351 @@
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use phf::phf_set;
+use serde::Deserialize;
+
+use crate::{context::LintContext, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-jsdoc(empty-tags): Expects the void tags to be empty of any content.")]
+#[diagnostic(severity(warning), help("`@{1}` tag should be empty."))]
+struct EmptyTagsDiagnostic(#[label] Span, String);
+
+#[derive(Debug, Default, Clone)]
+pub struct EmptyTags(Box<EmptyTagsConfig>);
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Expects the following tags to be empty of any content:
+    /// - `@abstract`
+    /// - `@async`
+    /// - `@generator`
+    /// - `@global`
+    /// - `@hideconstructor`
+    /// - `@ignore`
+    /// - `@inner`
+    /// - `@instance`
+    /// - `@override`
+    /// - `@readonly`
+    /// - `@inheritDoc`
+    /// - `@internal`
+    /// - `@overload`
+    /// - `@package`
+    /// - `@private`
+    /// - `@protected`
+    /// - `@public`
+    /// - `@static`
+    ///
+    /// ### Why is this bad?
+    /// The void tags should be empty.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Passing
+    /// /** @async */
+    ///
+    /// /** @private */
+    ///
+    /// // Failing
+    /// /** @async foo */
+    ///
+    /// /** @private bar */
+    /// ```
+    EmptyTags,
+    restriction
```

Comment:
I'd go for `correctness` because the text is useless.

But we need to add a `--jsdoc-plugin` flag cli first.

Reply:
> I'd go for correctness 

OK, I'll fix the category.

- - - 

> we need to add a --jsdoc-plugin flag cli first.

For this,

> I'll send a PR to make this plugin public after confirming that a few more rules can be implemented without any problems.

Please see also https://github.com/oxc-project/oxc/pull/2642